### DISCUSSION TITLE
Customizable auto-profile / setting engager

### DISCUSSION THREAD

**Discussion by Demitrix2k** [seerge/g-helper#2809] [Ideas]
Hey again, so quick context to explain the logic for my suggestion.
After having tried out various methods of having a custom fan curve, i realize and conclude, it doesn´t matter if you use Armory Crate or G-helper, the XG Mobile has an annoying issue where the fan rpm will alternate every second by 500 rpm by setting a custom fan curve, which can be annoying to listen to.

I know of course i can just toggle the custom fan curve on and off manually, my suggestion is this: an option to auto-enable custom fan curve for a profile by setting a custom temperature target trigger which activates it, and having a customizable deactivation, basically an automatic override of sorts.

so random example:
Ability to set 
"At 75c or higher, auto-enable custom fan curve for active profile"

Ability to set when it will deactivate custom fan curve for active profile, example: 
"at 65c or lower, and if set, x amount of minutes has passed and the gpu has been below specified temperature, deactivate custom fan curve"

I realize this is niche, if i wasn't so annoyed by my xg mobile, if only it could be normal like every other fan in my house, i wouldn't make this suggestion.

Thank you for your time.

**Comment by seerge** [comment]
@Demitrix2k hello, you are right - this is super niche thing. In most of the cases you can't just turn ON or OFF custom fan curves. There is a certain order of settings that app sets.

1. Base profile (0 balanced, 1 turbo, 2 silent) - this resets any previous custom settings
2. Custom fan curves (optional)
3. Custom power limits (optional, mostly don't work w/o custom fan curves being set along them)

So to turn custom fan settings ON and OFF app would have to re-set whole package of existing settings including base mode.

If you really want to do that I would just advice to create 2 profiles, one with custom settings, another one with defaults, and just switch between them using hotkeys.

Furthermore - you can even try to automate switching between this profiles using something like Autohotkey (that will monitor temp and emit one or another hotkey) and [existing keybindings](https://github.com/seerge/g-helper?tab=readme-ov-file#%EF%B8%8F-keybindings) in the app :)

**Reply by Demitrix2k** [reply]
Appreciate this advice, i will pursue this option further, thank you.
